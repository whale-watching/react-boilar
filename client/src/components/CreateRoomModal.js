import React from 'react';
import Modal from 'react-modal';
import PropTypes from 'prop-types';
import { socketEmit } from '../helpers/socketEvents';

class CreateRoomModal extends React.Component {
  constructor() {
    super();

    this.state = {
      error: null,
    };

    this.createRoom = (e) => {
      e.preventDefault();

      const roomName = e.target.elements.roomName.value.trim();
      const password = e.target.elements.password.value.trim();

      if (!roomName) {
        return this.setState({ error: 'You must enter room name' });
      }

      socketEmit.joinRoom(roomName, password);

      e.target.elements.roomName.value = '';
      e.target.elements.password.value = '';

      this.props.onRequestClose();
    };
  }

  render() {
    return (
      <Modal
        className="create-room-modal"
        isOpen={this.props.isOpen}
        onRequestClose={this.props.onRequestClose}
      >
        <form onSubmit={this.createRoom}>
          <h3>Create new room</h3>
          <p className="error">{this.state.error}</p>
          <p>Room name</p>
          <input type="text" name="roomName" maxLength="20" autoFocus autoComplete="off" />
          <p>Password (optional)</p>
          <input type="password" name="password" />
          <button type="submit" className="button-text">Create</button>
        </form>
      </Modal>
    );
  }
}

CreateRoomModal.propTypes = {
  isOpen: PropTypes.bool.isRequired,
  onRequestClose: PropTypes.func.isRequired,
};

export default CreateRoomModal;
