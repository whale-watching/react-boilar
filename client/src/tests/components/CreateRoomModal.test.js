import React from 'react';
import { shallow } from 'enzyme';
import CreateRoomModal from '../../components/CreateRoomModal';
import createRoomSubmit from '../data/createRoomSubmit';

describe('<CreateRoomModal', () => {
  it('renders CreateRoomModal correctly', () => {
    const wrapper = shallow(<CreateRoomModal isOpen={true} onRequestClose={() => { }}/>);

    expect(wrapper).toMatchSnapshot();
  });

  it('renders error if no room name is provided', () => {
    const wrapper = shallow(<CreateRoomModal isOpen={true} onRequestClose={() => { }} />);

    wrapper.find('form').simulate('submit', createRoomSubmit);

    expect(wrapper.state('error')).toBe('You must enter room name');
  });
});