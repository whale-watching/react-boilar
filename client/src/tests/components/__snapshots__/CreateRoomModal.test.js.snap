// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateRoomModal renders CreateRoomModal correctly 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="create-room-modal"
  closeTimeoutMS={0}
  isOpen={true}
  onRequestClose={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <form
    onSubmit={[Function]}
  >
    <h3>
      Create new room
    </h3>
    <p
      className="error"
    />
    <p>
      Room name
    </p>
    <input
      autoComplete="off"
      autoFocus={true}
      maxLength="20"
      name="roomName"
      type="text"
    />
    <p>
      Password (optional)
    </p>
    <input
      name="password"
      type="password"
    />
    <button
      className="button-text"
      type="submit"
    >
      Create
    </button>
  </form>
</Modal>
`;
